src/util/cable_climate_type_mod.F90

```diff
@@ -1,55 +1,364 @@
+  REAL, POINTER :: mtemp_max           (:)   
+    !! maximum monthly temperature                                  
+  REAL, POINTER :: qtemp_max           (:)   
+    !! mean temperature of the warmest quarter (so far this year)   
+  REAL, POINTER :: mtemp_min20         (:)   
+    !! minimum monthly temperature, averaged over 20 y              
+  REAL, POINTER :: mtemp_max20         (:)   
+    !! maximum monthly temperature, averaged over 20 y              
+  REAL, POINTER :: atemp_mean          (:)   
+    !! annual average temperature                                   
+  REAL, POINTER :: AGDD5               (:)                                                                  
+  REAL, POINTER :: GDD5                (:)   
+    !! growing degree day sum relative to 5deg base temperature     
+  REAL, POINTER :: AGDD0               (:)                                                                  
+  REAL, POINTER :: GDD0                (:)   
+    !! growing degree day sum relative to 0deg base temperature     
+  REAL, POINTER :: alpha_PT            (:)   
+    !! ratio of annual evap to annual PT evap                       
+  REAL, POINTER :: evap_PT             (:)   
+    !! annual PT evap [mm]                                          
+  REAL, POINTER :: aevap               (:)   
+    !! annual evap [mm]                                             
+  REAL, POINTER :: alpha_PT20          (:)                                                                  
+  REAL, POINTER :: qtemp_max_last_year (:)   
+    !! mean temperature of the warmest quarter (last calendar year) 
+
+  REAL, POINTER :: mtemp_min_20        (:,:) 
+    !! minimum monthly temperatures for the last 20 y   
+  REAL, POINTER :: mtemp_max_20        (:,:) 
+    !! maximum monthly temperatures for the last 20 y   
+  REAL, POINTER :: dtemp_31            (:,:) 
+    !! daily temperature for the last 31 days           
+  REAL, POINTER :: dmoist_31           (:,:) 
+    !! daily moisture availability for the last 31 days 
+  REAL, POINTER :: alpha_PT_20         (:,:) 
+    !! priestley Taylor Coefft for last 20 y            
+  REAL, POINTER :: dtemp_91            (:,:) 
+    !! daily temperature for the last 91 days           
 
 END TYPE climate_type
 
-!Instantiation:
-TYPE(climate_type) :: climate_cbl
+CONTAINS
+
+SUBROUTINE alloc_climate_type(climate, mp)
+!* Description:
+! Allocate and initialise arrays in the climate_data type
+
+USE grid_constants_mod_cbl,   ONLY: mf               ! # leaves (sunlit/shaded)
+USE grid_constants_mod_cbl,   ONLY: nsl              ! # soil layers                
+USE grid_constants_mod_cbl,   ONLY: niter            ! number of iterations for za/L
+
+IMPLICIT NONE
+
+TYPE(climate_data_type), INTENT(INOUT) :: climate
+INTEGER, INTENT(IN) :: mp
+!! Number of tiles/patches in the CABLE simulation
+
+ALLOCATE( climate% chilldays   (1) )
+ALLOCATE( climate% iveg        (1) )
```

Comment:
Why are we allocating everything to size 1? Shouldn't it be `mp`?

Reply:
you're right. This is just taken out of JAC, where we dont actually use them. For offline we need mp allocation.  Including #def UM_CBL - hangon I can use cable_rutime%

Reply:
@ccarouge DONE

Reply:
You've put it in ACCESS3-JULES but not here. I've copied over.